src/main/blackbox/blackbox_virtual.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <dirent.h>
+#include <common/maths.h>
+
+#define LOGFILE_PREFIX "LOG"
+#define LOGFILE_SUFFIX "BFL"
+
+static FILE *blackboxVirtualFile = NULL;
+static int32_t largestLogFileNumber = 0;
+bool blackboxVirtualOpen(void)
+{
+    DIR *dir = opendir(".");
```

Comment:
Add NULL check for opendir to prevent SEGFAULT

```suggestion
    DIR *dir = opendir(".");
    if (!dir) {
        return false; // Failed to open directory
    }

```

Reply:
Yes, this error maybe low probability, but why not?...